### ISSUE TITLE
Error while saving changes in the diagram (Flow Modeler screen)

### ISSUE THREAD

**Issue by Muddasar** [flowsforapex/apex-flowsforapex#697]
Hello,

I am facing an issue while saving my changes in the flow modeler screen:

_Update:
Oracle APEX 23.1.0
Flow For APEX 23.1
Oracle Database 19c_

### Issue: 
**1 error has occurred
Diagram could not be parsed.
Please review your diagram to ensure that it is supported.**

![image](https://github.com/flowsforapex/apex-flowsforapex/assets/7570845/20a9b819-2a86-4c9b-a7ed-5df5a63ccc6b)

Until now, it was working fine, my all the processes were also working fine. But now if I am trying to make new changes, I am getting this error.

If I try to start a new process or try to complete any existing step, I am getting following error as:

**1 error has occurred
No Next Step Found on subflow 178. Check your process diagram.**

![image](https://github.com/flowsforapex/apex-flowsforapex/assets/7570845/68ebeac0-a357-4379-adfb-536e7599f1f8)

Please note that, when I tried to look into the database tables, data from some of the tables is deleted (e.g. Data related to Flow Connections in FLOW_CONNECTIONS is not available anymore, please have a look at the screenshot attached below).

![Issue while saving FFA](https://github.com/flowsforapex/apex-flowsforapex/assets/7570845/e2862b0b-2a41-477f-a15b-d2a63cba648f)

I would like to request an urgent help here, as my application is ready to move in to Production environment, and in the last minutes I am facing this issue.

Best Regards,
Muddasar

**Comment by rallen2010** [comment]
As it looks like you didn't complete the bug template, please could you supply the following info:
- APEX version
- Flows for APEX version
- Oracle database version
- Have you Linted your BPMN diagram in the modeler?  ( click on the lozenge at the bottom of the diagram that reads 0 Errors, 0 warnings)?

**Comment by Muddasar** [comment]
> As it looks like you didn't complete the bug template, please could you supply the following info:
> 
> * APEX version
> * Flows for APEX version
> * Oracle database version
> * Have you Linted your BPMN diagram in the modeler?  ( click on the lozenge at the bottom of the diagram that reads 0 Errors, 0 warnings)?

Thank you very much for your consideration and reply dear Rallen.

Here you go, sorry for missing earlier:

Oracle APEX 23.1.0
Flow For APEX 23.1
Oracle Database 19c

Plus, when I click on Errors / Warning section, it shows warnings for missing label of gateways only.

Please guide me more, how can I fix this issue.

Regards,
Muddasar

**Comment by rallen2010** [comment]
So this looks like a problem with the parser - either a parser bug or something unsupported in your model.  
When a model is saved and parsed, it does delete the previous parsed representation of the diagram - so having flow_connections not contain records for the diagram is to be expected...
So I can see more of what's going on, could you do the following:

1. Turn on logging in the parser.  If you look in the flow_configuration table, you need to have an entry for `parser_log_enabled` set to `true`.  If there is an existing entry, please set the value to `true`.   If there is no entry for `parser_log_enabled`, then please insert one and set it to `true`.   You can use the APEX Object Browser to do this...

2. Go to the Flow Modeller in the Flows for APEX app as a developer, then on the Developer Menu at the bottom of your screen, Select Debug > Enable Debug > App Trace level.
3. Save your model again - which I assume will fail.
4. Can you then send me the BPMN (export as BPMN), the APEX Debug output and the parser log.  Either post it on here or send it to my email if that's OK.

**Comment by Muddasar** [comment]
Dear RAllen,

Thank you for your support.

I have followed the first three steps (i.e. set parser_log_enabled = true, enabled App Trace Level debugging and tried to save the model, yes it failed to save).

Here, I am sending the BPMN & APEX Debug in your email, as the bpmn format is not supported here to be attached. But there's no data in FLOW_PARSER_LOG table.

[Debug Message Data.xlsx](https://github.com/flowsforapex/apex-flowsforapex/files/13627706/Debug.Message.Data.xlsx)

![image](https://github.com/flowsforapex/apex-flowsforapex/assets/7570845/02c0cfec-ee56-4a75-8384-ba4e250703f8)

Please check your email for BPMN file.

Thanks & Regards,
Muddasar

**Comment by Muddasar** [comment]
Dear R Allen,

Good afternoon.

Please find the attached bpmn file.

Please note that, there's no log available in FLOW_PARSER_LOG table.

Looking forward to hearing from you.

Regards,
Muddasar Amin

Sr. Software Engineer

Alumni NUST-SEECS
Islamabad Pakistan

+96898978096

**Comment by Muddasar** [comment]
Thank you very much for your help, after a detailed analysis of flow_bpmn_parser_pkg.parse( pi_dgrm_id => :dgrm_id ); I got to know that, during enhancement of FFA application for my custom requirements I had referenced FLOW_OBJECTS.OBJT_ID into one of my local DB tables and it was not allowing to delete the rows.

I have disabled that foreign key and now it's working perfectly fine. Apologies for the inconvenience.
